src/common/interProcessMessageSender.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as net from "net";
+import * as Q from "q";
+import {Log} from "./log/log";
+import {LogLevel} from "./log/logHelper";
+
+import {ExtensionMessage, MessageWithArguments, ErrorMarker} from "./extensionMessaging";
+
+export interface IInterProcessMessageSender {
+    sendMessage(message: ExtensionMessage, args?: any[]): Q.Promise<any>;
+}
+
+// TODO: Refactor this class to make ExtensionMessage a generic parameter instead
+
+/**
+ * Sends messages to the extension.
+ */
+export class InterProcessMessageSender implements InterProcessMessageSender {
+    constructor(private serverPath: string) {
```

Comment:
Same here. Please add an `/* empty */` comment. Also, what is the deal with the parameter? Is this for testing only?